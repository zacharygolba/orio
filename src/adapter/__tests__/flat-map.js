// @flow

import FlatMap from '../flat-map'
import * as iter from '../../'

let subj

beforeEach(() => {
  const { producer } = iter.from([1, 2, 3])
  subj = new FlatMap(producer, item => iter.repeat(item).take(2))
})

test('#@@iterator()', () => {
  for (const item of subj) {
    expect(item).toMatchSnapshot()
  }
})

test('#next()', () => {
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
})

test('#sizeHint()', () => {
  expect(subj.sizeHint()).toEqual(3)

  // eslint-disable-next-line no-empty, no-unused-vars
  for (const _ of subj) {
  }

  expect(subj.sizeHint()).toEqual(6)
})
