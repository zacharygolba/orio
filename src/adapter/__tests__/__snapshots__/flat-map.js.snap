// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#@@iterator() 1`] = `1`;

exports[`#@@iterator() 2`] = `1`;

exports[`#@@iterator() 3`] = `2`;

exports[`#@@iterator() 4`] = `2`;

exports[`#@@iterator() 5`] = `3`;

exports[`#@@iterator() 6`] = `3`;

exports[`#next() 1`] = `
Object {
  "done": false,
  "value": 1,
}
`;

exports[`#next() 2`] = `
Object {
  "done": false,
  "value": 1,
}
`;

exports[`#next() 3`] = `
Object {
  "done": false,
  "value": 2,
}
`;

exports[`#next() 4`] = `
Object {
  "done": false,
  "value": 2,
}
`;

exports[`#next() 5`] = `
Object {
  "done": false,
  "value": 3,
}
`;

exports[`#next() 6`] = `
Object {
  "done": false,
  "value": 3,
}
`;

exports[`#next() 7`] = `
Object {
  "done": true,
  "value": undefined,
}
`;
